### PR TITLE
Make tags there own resourced, allow hiera created integrations, setup check_feq

### PR DESCRIPTION
**Pull request by cwood**
These are a bunch of changes we have made to the puppet module.

1. Make tags there own resources. This allows use to use hiera to define tags and look up the chain.
2. Purge integrations that are not managed via puppet. We use this so we can make sure only the integrations we define are in use.
3. I think the rest are just random stuff we have used to fix / setup better hiera flow.

### CODE REVIEW

**Comment by cwood** [comment]
@truthbk This is the new version. It should be fine. Please keep me posted. I dont want this to be like the integration PR where it sat for like 6 months. There are some usefull features in this PR. That I would think the community would like. The ability to define a tag as a resource makes it a lot more flexable for us.

**Comment by cwood** [comment]
Thanks for taking over this. Hopefully the community finds this useful.

**Comment by truthbk** [comment]
@cwood to the contrary, thank you for your constant contributions. I'm trying to make this include a few new features that were merged, and ensure we don't break anything. Thanks again. I might ask for a review once we're done. Hopefully we can merge this soon.

**Comment by truthbk** [comment]
Cool, I think we're green now. I'll have to run a few tests... I'm hoping we can merge this before Friday, and if no backward incompatibilities are found we'll have this in the next module release.

@cwood heads-up on a force push to your branch.

File: manifests/init.pp [hunk 1]
```diff
@@ -52,6 +52,8 @@
 #   $statsd_forward_port
 #       Set the value of the statsd_forward_port varable. Used to forward all
 #       statsd metrics to another host.
+#   $check_freq
+#       Sets how often checks should be run. By default, set to 15s
```
Comment (sjenriquez) [inline manifests/init.pp]: While this is valid, we don't recommend changing the `check_freq`. I think we should remove this parameter.

File: manifests/init.pp [hunk 2]
```diff
@@ -200,6 +204,7 @@
   $statsd_forward_host = '',
   $statsd_forward_port = '',
   $statsd_histogram_percentiles = '0.95',
+  $check_freq = '',
```
Comment (sjenriquez) [inline manifests/init.pp]: See above

File: templates/datadog_footer.conf.erb [hunk 3]
```diff
@@ -124,6 +60,14 @@ use_curl_http_client: <%= @use_curl_http_client %>
...
 <% end -%>
 
+# Sets how often checks should be run.
+<% if @check_freq.empty? -%>
+# check_freq: 15
+<% else -%>
+check_freq: <%= @check_freq %>
+<% end -%>
+
+
```
Comment (sjenriquez) [inline templates/datadog_footer.conf.erb]: See comment above regarding `check_freq`

**Comment by sjenriquez** [comment]
Left a comment about `check_freq`, but otherwise excellent changes here!

**Comment by truthbk** [comment]
Thanks @sjenriquez, great catch on that one. Will merge if tests are green.

File: manifests/init.pp (hunk 1 above)
Comment (cwood) [inline manifests/init.pp]: This will break us if we dont have that. If its exposed by the dd-agent and documented why cant it be set by the module?

**Comment by cwood** [comment]
I put in the comment that we do use check_freq so that isnt really great for us. Ill have to double check what we set it too. But id rather keep it then remove it.

**Comment by truthbk** [comment]
Going to merge this, can we revisit `check_freq` in a separate PR/issue? Thanks!

**Comment by cwood** [comment]
Its no worries thanks for working on this. Glad this got merged in :)

**Comment by flyinprogrammer** [comment]
This broke the `extra_template` parameter 😢

**Pull request merged**